tedana/io.py

```diff
@@ -501,6 +504,195 @@ def filewrite(data, filename, ref_img, gzip=False, copy_header=True):
     return name
 
 
+def writefigures(ts, mask, comptable, mmix, n_vols,
+                 acc, rej, midk, empty, ref_img):
+    """
+    Creates some really simple plots useful for debugging
+
+    Parameters
+    ----------
+    ts : (S x T) array_like
+        Time series from which to derive ICA betas
+    mask : (S,) array_like
+        Boolean mask array
+    comptable : (N x 5) array_like
+        Array with columns denoting (1) index of component, (2) Kappa score of
+        component, (3) Rho score of component, (4) variance explained by
+        component, and (5) normalized variance explained by component
+    mmix : (C x T) array_like
+        Mixing matrix for converting input data to component space, where `C`
+        is components and `T` is the same as in `data`
+    n_vols : :obj:`int`
+        Number of volumes in original time series
+    acc : :obj:`list`
+        Indices of accepted (BOLD) components in `mmix`
+    rej : :obj:`list`
+        Indices of rejected (non-BOLD) components in `mmix`
+    midk : :obj:`list`
+        Indices of mid-K (questionable) components in `mmix`
+    empty : :obj:`list`
+        Indices of ignored components in `mmix`
+    ref_img : :obj:`str` or img_like
+        Reference image to dictate how outputs are saved to disk
+
+    """
+
+    # regenerate the beta images
+    ts_B = model.get_coeffs(ts, mmix, mask)
+    ts_B = ts_B.reshape(ref_img.shape[:3] + ts_B.shape[1:])
+    # Mask out zeros
+    ts_B = np.ma.masked_where(ts_B == 0, ts_B)
+
+    # Get repitition time from ref_img
+    tr = ref_img.header.get_zooms()[-1]
+
+    # Start making some really ugly pluts
+    import os
+    if not os.path.exists('./figures'):
+        os.mkdir('figures')
+
+    os.chdir('./figures')
+
+    # This precalculates the Hz for the fft plots
+    Fs = 1.0/tr
+    # resampled frequency vector
+    f = Fs * np.arange(0, n_vols // 2 + 1) / n_vols
+
+    # Create indices for 6 cuts, based on dimensions
+    xdim = ts_B.shape[0]
+    xcut = int(xdim/6)
+
+    ydim = ts_B.shape[1]
+    ycut = int(ydim/6)
+
+    zdim = ts_B.shape[2]
+    zcut = int(zdim/6)
+
+    for compnum in range(0, mmix.shape[1], 1):
```

Comment:
I wonder if we should make one list at the top where we map every component to its classification. Then we could clean up a lot of the subsequent code substantially. Do we already make this in the main workflow, anywhere ?

Reply:
the components are classified, and that is passed as an argument - and this is used to set line color.    I do get a strong feeling that it can be further cleaned up, so I'll keep thinking on it. Any further advice is super appreciated.